lib/chef/resource/windows_certificate.rb

```diff
@@ -38,6 +38,10 @@ class WindowsCertificate < Chef::Resource
       property :pfx_password, String,
         description: "The password to access the source if it is a pfx file."
 
+      property :pfx_properties, Integer,
+        description: "The dwFlags value desired for PFX storage. Automatically assigned based on the user_store property if omitted. See https://docs.microsoft.com/en-us/windows/desktop/api/wincrypt/nf-wincrypt-pfximportcertstore PFXImportCertStore function for available values.",
+        default: lazy { user_store ? 0x00001000 : 0x00000020 }
```

Comment:
This is a really useful property to add, but I think this doesn't address it in a way that we'd want to merge it. Many of those pfx properties are things that our users would want to be able to set, but they don't want to have to add up hex values to determine the values they want to set. This will need to be something where the user set the properties in a friendly value and we passed handled passing in the proper hex value.